src/components/user/system-agent.gel.repository.ts

```diff
@@ -12,15 +12,15 @@ export class SystemAgentGelRepository extends SystemAgentRepository {
   }
 
   protected async upsertAgent(name: string, roles?: readonly Role[]) {
-    const query = edgeql(`
-      select (
-        (select SystemAgent filter .name = <str>$name) ??
-        (insert SystemAgent {
-          name := <str>$name,
-          roles := array_unpack(<optional array<Role>>$roles)
-        })
-      ) {*}
-    `);
-    return await this.db.run(query, { name, roles });
+    const upserted = e.op(
+      e.select(e.SystemAgent, () => ({ filter_single: { name } })),
+      '??',
+      e.insert(e.SystemAgent, { name, roles }),
+    );
+    const query = e.select(upserted, (agent) => ({
+      __typename: e.str('SystemAgent' as const),
```

Comment:
I had to rewrite this query to use the query builder, because EdgeQL doesn't understand string literals, so even though I could write the constant "SystemAgent" it would still be widened to a TS `string`, by the EdgeQL server.